src/server/index.ts

```diff
@@ -65,6 +66,7 @@ interface ServerOpts {
   authenticateCredentials?: ServerTypes.AuthenticateCredentials;
   generateCredentials?: ServerTypes.GenerateCredentials;
   https?: HttpsOptions;
+  socketOpts?: SocketOptions;
```

Comment:
This should probably be a part of the transport option on line 65.  The server doesn't have to use sockets, so we shouldn't bake in options that assume that it does.

Reply:
That works for me too. I updated this pull request to expose SocketIO in the package so that clients can configure it themselves and pass it through to the Server constructor.

Reply:
Oh, I saw that this is already exposed in the main package. Closing this PR.

Reply:
Note that with the SocketIO transport exposed, it would also be possible to remove the `https` and `authenticateCredentials` arguments to the Server constructor. Let me know if you'd like me to make this change and update the documentation accordingly.

Reply:
@c-w I think it’s OK to leave these here for now. It’s especially nice to have `authenticateCredentials` alongside `generateCredentials` without needing to manually instantiate the transport class.